tests/e2e/App.spec.ts

```diff
@@ -0,0 +1,306 @@
+  await page.getByRole("button", { name: "Submit Command" }).click();
+
+  await expect(page.locator('[data-index="history-command-0"]')).toContainText(
+    "Error: dataset not loaded for search"
+  );
+});
+
+test("i can do EVERYTHING: view and search before loading (gets errors), load a nonexistent CSV file, load an existing file, view it, and search existent AND nonexistent data with column name and index", async ({
+  page,
+}) => {
+  // unloaded view
+  await page.getByLabel("Command input").fill("view");
+  await page.getByRole("button", { name: "Submit Command" }).click();
+
+  await expect(page.locator('[data-index="history-command-0"]')).toContainText(
+    "Error: dataset not loaded for view"
+  );
+
+  // unloaded search
+  await page.getByLabel("Command input").fill("search 1 hi");
+  await page.getByRole("button", { name: "Submit Command" }).click();
+
+  await expect(page.locator('[data-index="history-command-1"]')).toContainText(
+    "Error: dataset not loaded for search"
+  );
+
+  // load malformed filepath
+  await page.getByLabel("Command input").fill("load_file hii");
+  await page.getByRole("button", { name: "Submit Command" }).click();
+
+  await expect(page.locator('[data-index="history-command-2"]')).toContainText(
+    "Error: File not found at hii"
+  );
+
+  // load formed filepaht
+  await page.getByLabel("Command input").fill("load_file path2");
+  await page.getByRole("button", { name: "Submit Command" }).click();
+
+  await expect(page.locator('[data-index="history-command-3"]')).toContainText(
+    "Loaded dataset from path2"
+  );
+
+  // view loaded file
+  await page.getByLabel("Command input").fill("view");
+  await page.getByRole("button", { name: "Submit Command" }).click();
+
+  await expect(page.getByRole("table")).toBeVisible();
+
+  // search loaded file in wrong column
+  await page.getByLabel("Command input").fill("search Header2 Data6");
+  await page.getByRole("button", { name: "Submit Command" }).click();
+
+  await expect(page.locator('[data-index="history-command-5"]')).toContainText(
+    "No results found for"
+  );
+
+  // search loaded file in correct column with NAME
+  await page.getByLabel("Command input").fill("search Header3 Data6");
+  await page.getByRole("button", { name: "Submit Command" }).click();
+
+  await expect(page.locator('[data-index="history-command-6"]')).toContainText(
+    "Data4 Data5 Data6"
+  );
+  await expect(page.locator('[data-index="history-command-7"]')).toContainText(
+    "Data4 Data7 Data6"
+  );
+
+  // search loaded file in correct column with INDEX
+  await page.getByLabel("Command input").fill("search 2 Data6");
+  await page.getByRole("button", { name: "Submit Command" }).click();
+
+  await expect(page.locator('[data-index="history-command-8"]')).toContainText(
+    "Data4 Data5 Data6"
+  );
+  await expect(page.locator('[data-index="history-command-9"]')).toContainText(
+    "Data4 Data7 Data6"
+  );
```

Comment:
these are very great tests that cover a broad usage of the program! i think one additional state to think about is signing out, and ensuring that the history is reset